src/inspect_ai/util/_sandbox/docker/docker.py

```diff
@@ -315,12 +307,26 @@ async def write_file(self, file: str, contents: str | bytes) -> None:
             project=self._project,
         )
 
-        res_cp = await self.exec(["cp", "-T", "--", container_tmpfile, file])
+        parent = Path(file).parent.as_posix()
```

Comment:
Note that for doing path resolution that is targeting the sandbox, we should use `PurePosixPath` (as that we can actually use on Windows). We actually had to do that in `exec()` already to fix a bug on Windows.